app/components/channel_drawer/channels_list/list/list.js

```diff
@@ -1,157 +1,122 @@
-
 import {General} from 'mattermost-redux/constants';
+import {debounce} from 'mattermost-redux/actions/helpers';
+import EventEmitter from 'mattermost-redux/utils/event_emitter';
 
 import ChannelItem from 'app/components/channel_drawer/channels_list/channel_item';
-import UnreadIndicator from 'app/components/channel_drawer/channels_list/unread_indicator';
+import ChannelUnreadItem from 'app/components/channel_drawer/channels_list/channel_unread_item';
+import UnreadIndicator, {SET_WITDH, SHOULD_SHOW_EVENT} from 'app/components/channel_drawer/channels_list/unread_indicator';
+import {wrapWithPreventDoubleTap} from 'app/utils/tap';
+import {changeOpacity} from 'app/utils/theme';
 
-class List extends Component {
+class List extends PureComponent {
     static propTypes = {
         canCreatePrivateChannels: PropTypes.bool.isRequired,
-        channels: PropTypes.object.isRequired,
-        channelMembers: PropTypes.object,
-        currentChannel: PropTypes.object,
+        directChannelIds: PropTypes.array.isRequired,
+        favoriteChannelIds: PropTypes.array.isRequired,
         intl: intlShape.isRequired,
         navigator: PropTypes.object,
         onSelectChannel: PropTypes.func.isRequired,
+        publicChannelIds: PropTypes.array.isRequired,
+        privateChannelIds: PropTypes.array.isRequired,
         styles: PropTypes.object.isRequired,
-        theme: PropTypes.object.isRequired
-    };
-
-    static defaultProps = {
-        currentChannel: {}
+        theme: PropTypes.object.isRequired,
+        unreadChannelIds: PropTypes.array.isRequired
     };
 
     constructor(props) {
         super(props);
-        this.firstUnreadChannel = null;
-        this.state = {
-            dataSource: this.buildData(props),
-            showAbove: false
-        };
 
         MaterialIcon.getImageSource('close', 20, this.props.theme.sidebarHeaderTextColor).then((source) => {
             this.closeButton = source;
         });
     }
 
-    shouldComponentUpdate(nextProps, nextState) {
-        return !deepEqual(this.props, nextProps, {strict: true}) || !deepEqual(this.state, nextState, {strict: true});
-    }
-
-    componentWillReceiveProps(nextProps) {
-        this.setState({
-            dataSource: this.buildData(nextProps)
-        }, () => {
-            if (this.refs.list) {
-                this.refs.list.recordInteraction();
-                this.updateUnreadIndicators({
-                    viewableItems: Array.from(this.refs.list._listRef._viewabilityHelper._viewableItems.values()) //eslint-disable-line
-                });
-            }
-        });
+    componentDidUpdate(prevProps) {
+        if (prevProps.unreadChannelIds !== this.props.unreadChannelIds && this.refs.list) {
+            this.refs.list.recordInteraction();
+            this.updateUnreadIndicators({
+                viewableItems: Array.from(this.refs.list._wrapperListRef._listRef._viewabilityHelper._viewableItems.values()) //eslint-disable-line
```

Comment:
This looks like it's going to break as soon as we update RN. Is there not a way to do this with external APIs?

Reply:
Oh, this didn't actually change from how it worked before. Never mind then.